lms/djangoapps/courseware/user_state_client.py

```diff
@@ -166,16 +209,40 @@ def set_many(self, username, block_keys_to_state, scope=Scope.user_state):
                 },
             )
 
+            num_fields_before = num_fields_after = num_new_fields_set = len(state)
+            num_fields_updated = 0
             if not created:
                 if student_module.state is None:
                     current_state = {}
                 else:
                     current_state = json.loads(student_module.state)
+                num_fields_before = len(current_state)
                 current_state.update(state)
+                num_fields_after = len(current_state)
```

Comment:
if this is something we are planning on taking out later it would be better if we could isolate the metric generation from the actual state modification in these calls. It would make the code that is acting upon the state more clear and what code is simply cruft if you are not interested in the metrics